File: SriEshwar/Batch-2/01-11-23/TOH.java
public class TOH {
    public static void main(String[] args) {
        int n = 3;
        char src ='A', des = 'C', aux = 'B';
        toh(n,src,des,aux);
    }
    private static void toh(int n, char src, char des, char aux) {
        if(n<1) return;
        toh(n-1,src,aux,des);
        System.out.println(src+"->"+des);
        toh(n-1, aux, des, src);
    }
}

File: Python/Toh.py
def toh(n,src,aux,des):
    if n <= 0:
        return
    toh(n-1,src,des,aux)
    print(src,"->",des)
    toh(n-1,aux,src,des)
def main():
    n = 3
    src,aux,des = 'A','B','C'
    toh(n,src,aux,des)
main()